src/loaders/agent.js

```diff
@@ -46,7 +46,9 @@ export class Agent extends AgentBase {
     // Future work is being planned to evaluate removing this behavior from the backend, but for now we must ensure this call is made
     this.desiredFeatures.add(PageViewEvent)
 
-    Object.assign(this, configure(this.agentIdentifier, options, options.loaderType || 'agent'))
+    gosNREUMInitializedAgents(this.agentIdentifier, this) // append this agent onto the global NREUM.initializedAgents
+    // ^Currently, this needs to happen before configure() since it will add the agent to the global that's looked up by set-config functions inside there.
```

Comment:
should we just put line 49 inside at the top of configure to eliminate the cross-dependency?  this will already exist in there now

Reply:
So the real problem in this case really is just that the first set of set___ calls done by configure won't add (update) those configurations on the class instance itself since they won't find anything on the global yet, should this line go at the bottom of the constructor.
It wouldn't be an issue of course if those calls were just passed the instance argument to which they can just update the property on the instance in the grand scheme of things.

But yes to your comment, this can just be moved up top.